Port of retirement snapshots

0xAurelius opened the pull request:
Still has some issues related to changed data types and fields being moved around/renamed, esp. the "token" field being gone, the feeAmount being the wrong datatype (int vs. decimal) and the "pool" field being `Bytes | null` instead of `string`

0xAurelius commented:
I'm still struggling to get this working properly, MCO2 retired stats on `CarbonMetrics` are only showing 0.001 retired (maybe a decimal issue?) and `carbonCreditSnapshots` are not being populated properly...

psparacino commented:
> I'm still struggling to get this working properly, MCO2 retired stats on `CarbonMetrics` are only showing 0.001 retired (maybe a decimal issue?) and `carbonCreditSnapshots` are not being populated properly...

are you taking into account the kilogram conversion? taking a look now

psparacino commented:
> I'm still struggling to get this working properly, MCO2 retired stats on `CarbonMetrics` are only showing 0.001 retired (maybe a decimal issue?) and `carbonCreditSnapshots` are not being populated properly...

which deployment are you working with? Spun up a test here: https://github.com/KlimaDAO/klima-subgraph/pull/328 for moss metrics and in the test at least seems to be ok...? the snapshots don't work for mco2 or all credits?

0xAurelius commented:
I was just using `dev-polygon-digital-carbon` on the Subgraph Studio.

Snapshots didn't seem to be populating at all (when I query them get 0 results)

psparacino commented:
> I was just using `dev-polygon-digital-carbon` on the Subgraph Studio.
> 
> Snapshots didn't seem to be populating at all (when I query them get 0 results)

i think i found the issue with snapshots, adjusted here: https://github.com/KlimaDAO/klima-subgraph/pull/328/files#diff-e60082d3bd2164efaef060d6749338805ed17c4d7625b3b844aca384fa75bcefR149-R163

the matchstick tests work for mco2Retired. On your dev deployment `4a35369` the `mco2Retired`. values look valid unless I'm missing something?

```graphql
query MyQuery {
  carbonMetrics(where: {mco2Retired_gt: "0"}) {
    id
    mco2Retired
  }
}
```

0xAurelius commented:
> > I was just using `dev-polygon-digital-carbon` on the Subgraph Studio.
> > Snapshots didn't seem to be populating at all (when I query them get 0 results)
> 
> i think i found the issue with snapshots, adjusted here:
> 
>     1. toHexString() needed: https://github.com/KlimaDAO/klima-subgraph/pull/328/files#diff-e60082d3bd2164efaef060d6749338805ed17c4d7625b3b844aca384fa75bcefR219
> 
>     2. used return from saveKlimaRetire: https://github.com/KlimaDAO/klima-subgraph/pull/328/files#diff-e60082d3bd2164efaef060d6749338805ed17c4d7625b3b844aca384fa75bcefR149-R163
> 
> 
> I pushed all of the above to `dev` which is syncing
> 
> the matchstick tests work for mco2Retired. On your dev deployment `4a35369` the `mco2Retired`. values look valid unless I'm missing something?
> 
> ```graphql
> query MyQuery {
>   carbonMetrics(where: {mco2Retired_gt: "0"}) {
>     id
>     mco2Retired
>   }
> }
> ```

Nice catch! I just added in these two fixes and also merged in your matchstick test for MCO2 logic - thanks!

Will check the version indexing on dev when it's done

0xAurelius commented:
OK snapshots are populating now, but there is still an issue with decimals, for instance this query:
```
query Query {
  dailyKlimaRetirements (first:50, orderBy: timestamp, orderDirection: desc){
    id
    timestamp
    amount
    feeAmount
  }
}
```

Doesn't match formatting between the bridged and digital subgraphs:
![image](https://github.com/user-attachments/assets/6b241177-24da-49ec-880c-51b5e1ed7a7f)
![image](https://github.com/user-attachments/assets/4ca1b921-ceb7-4ec1-8044-ba750f78f507)


There does seem to be something up with MCO2 retired... I'm seeing values in a very wide range, these seem unreasonably big: ```query MyQuery {
  carbonMetrics(where: {mco2Retired_gt: "1000"}, orderBy: timestamp, orderDirection:desc) {
    id
    mco2Retired
    tco2Retired
    cco2Retired
    
  }
}```

![image](https://github.com/user-attachments/assets/d599c7db-1a23-4655-b3dd-af4f8d7a1f4d)


But if you change the filter to be `0` then you'll see the recent metrics entries have very small decimal amounts...

psparacino commented on polygon-digital-carbon/src/utils/KlimaRetire.ts:
```diff
@@ -15,8 +16,13 @@ export function saveKlimaRetire(
     klimaRetire.retire = retire
     klimaRetire.index = index
     klimaRetire.feeAmount = feeAmount
+    klimaRetire.feeAmountTonnes = toDecimal(feeAmount)
```
assuming this uses 18 decimals as default
